### PR TITLE
[SPARK-39598][PYTHON] Make *cache*, *catalog* in the python side support 3-layer-namespace

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
corresponding changes in pyspark to [CacheTable, isCached, UncacheTable, setCurrentCatalog, currentCatalog, listCatalogs](https://issues.apache.org/jira/browse/SPARK-39506)

1, add method `currentCatalog`, `setCurrentCatalog`, `listCatalogs`;
2, update the document of `CacheTable`, `isCached`, `UncacheTable`, and add a UT

### Why are the changes needed?
to support 3-layer-namespace

### Does this PR introduce _any_ user-facing change?
yes, new api added

### How was this patch tested?
added ut

### CODE REVIEW

**Comment by zhengruifeng** [comment]
All test passed except document build failure which also appears in other PRs.

cc @HyukjinKwon @cloud-fan @amaliujia

**Comment by dongjoon-hyun** [comment]
I made a fix here to recover master branch, @zhengruifeng . 
- https://github.com/apache/spark/pull/36987

**Review by cloud-fan — APPROVED** [review]

**Review by HyukjinKwon — COMMENTED** [review]
LGTM2

File: python/pyspark/sql/catalog.py [hunk 1]
```diff
@@ -79,6 +84,42 @@ def __init__(self, sparkSession: SparkSession) -> None:
         self._jsparkSession = sparkSession._jsparkSession
         self._jcatalog = sparkSession._jsparkSession.catalog()
 
+    def currentCatalog(self) -> str:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/catalog.py:87]: Just dropping a comment .. let's make sure these are seen in PySpark documentation properly.

Comment (zhengruifeng) [reply python/pyspark/sql/catalog.py:87]: let me check it.

**Pull request closed without merging**